.github/workflows/gradle-ethereum-tests.yml

```diff
@@ -0,0 +1,72 @@
+name: gradle
+
+on:
+  schedule:
+    - cron: "0 23 * * *"
+
+env:
+  CARGO_NET_GIT_FETCH_WITH_CLI: true
+
+jobs:
+  ethereum-tests:
+    runs-on: ubuntu-latest-128
+    steps:
+      - uses: webfactory/ssh-agent@v0.7.0
+        with:
+          ssh-private-key: |
+            ${{ secrets.CONSTRAINTS_SSH_KEY }}
+
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Set up JDK 21
+        uses: actions/setup-java@v3
+        with:
+          java-version: 21
+          distribution: temurin
+
+      - name: Install Rust
+        uses: actions-rust-lang/setup-rust-toolchain@v1
+
+      # The asset URL for the latest release can be found with:
+      # curl -L -H "Accept: application/vnd.github+json"  \
+      #     -H "Authorization: Bearer YOUR_GH_API_TOKEN" \
+      #     -H "X-GitHub-Api-Version: 2022-11-28" \
+      #     https://api.github.com/repos/ConsenSys/corset/releases/latest
+      # | jq '.assets[] | select(.name|endswith("x86_64-unknown-linux-musl.tar.gz")) | .url'
+      - name: Install Corset
+        run: RUSTFLAGS=-Awarnings cargo install --git ssh://git@github.com/ConsenSys/corset --tag v9.7.13 --locked --force
+#        run: |
+#          curl -L \
+#          -H "Accept: application/octet-stream" \
+#          -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+#          -H "X-GitHub-Api-Version: 2022-11-28" \
+#          -o corset.tar.gz \
+#          https://api.github.com/repos/Consensys/corset/releases/assets/151396061
+#          tar xzf corset.tar.gz
+#          mv corset $HOME
+#          echo $HOME >> $GITHUB_PATH
+
+      - name: Generate block chain reference tests
+        run:  ./gradlew :reference-tests:generateBlockchainReferenceTests -Dorg.gradle.parallel=true -Dorg.gradle.caching=true
+        env:
+          JAVA_OPTS: -Dorg.gradle.daemon=false -DZKEVM_BIN_ORIGINAL_PATH="zkevm-constraints/zkevm.bin"
+          CORSET_FLAGS: fields,expand,expand,expand
+
+      - name: generate zkevm.bin
+        run: cd ./zkevm-constraints; make zkevm.bin -B; cd ..
+
+      - name: Run a single reference test
+        run: ./gradlew :reference-tests:referenceTests --tests "net.consensys.linea.generated.blockchain.BlockchainReferenceTest_583"
+
+      - name: Run general reference tests
+        run: ./gradlew referenceGeneralStateTests
+        env:
+          JAVA_OPTS: -Dorg.gradle.daemon=false
+          CORSET_FLAGS: fields,expand,expand,expand
+
+      - name: Run all reference tests
+        run: ./gradlew referenceTests
```

Comment:
.`/gradlew referenceTestsThis` will run `referenceBlockchainTests` and  `referenceGeneralStateTests`  which makes the previous step `Run general reference tests` redundant.

So maybe remove the "Run all reference tests" and add a new step for `referenceBlockchainTests`

Reply:
thanks i remove it. I am not adding block chain tests are they are not working right now.